### PR TITLE
[RFC] Standardize Collective (AllGatherOp, AllReduceOp, AllToAllOp) ops to support variadic operand/result

### PR DESCRIPTION
**Pull request by abhigunj**
This RFC proposes standardizing collective (AllGatherOp, AllReduceOp, AllToAllOp) ops to  enable support for `multi-operand` and `multi-result`

Please review and provide you feedback.

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -714,40 +714,40 @@ defined as follows:
...
-| Label | Name                    | Type                                         | Constraints |
-|-------|-------------------------|----------------------------------------------|-------------|
-| (I1)  | `operand`               | tensor or per-tensor quantized tensor        | (C1), (C6)  |
-| (I2)  | `all_gather_dim`        | constant of type `si64`                      | (C1), (C6)  |
-| (I3)  | `replica_groups`        | 2-dimensional tensor constant of type `si64` | (C2-C4)     |
-| (I4)  | `channel_id`            | constant of type `si64`                      | (C5)        |
-| (I5)  | `use_global_device_ids` | constant of type `i1`                        | (C5)        |
+| Label | Name                    | Type                                                        | Constraints |
+|-------|-------------------------|-------------------------------------------------------------|-------------|
+| (I1)  | `operands`               | variadic number of tensors or per-tensor quantized tensors | (C1), (C6)  |
```
Comment (mlevesquedion) [inline docs/spec.md:726]: Nit, here and elsewhere: adjust the table so the columns are aligned.

**Review by mlevesquedion — APPROVED** [review]

File: docs/spec.md (hunk 1 above)
Comment (abhigunj) [reply docs/spec.md]: thank you, updated

**Review by ghpvnist — CHANGES_REQUESTED** [review]
Could you update the examples to showcase how this works?

**Review by sdasgup3 — APPROVED** [review]
LGTM once the remaining request related to providing  example is fixed.

**Review by GleasonK — APPROVED** [review]

**Review by ghpvnist — APPROVED** [review]

**Pull request merged**